### PR TITLE
Use refactored confgen for metrics

### PR DESCRIPTION
**Pull request by jotak**
Requires: https://github.com/netobserv/flowlogs-pipeline/pull/267

Opened as draft until the above is merged

cc @KalmanMeth there's a couple of little changes on top of what you added :

- Interface with confgen is reduced to only what is needed here: calling "ParseDefinition" and "GenerateTruncatedConfig". So we don't need anymore the intermediate temporary files
- Propagate errors to caller

### CODE REVIEW

File: controllers/flowlogspipeline/flp_test.go [hunk 1]
```diff
@@ -560,24 +580,21 @@ func TestPipelineTraceStage(t *testing.T) {
...
 
-//go:embed test_metrics_definitions
-var TestFlpMetricsConfig embed.FS
-
 func TestMergeMetricsConfigurationNoIgnore(t *testing.T) {
 	assert := assert.New(t)
 
 	flp := getFLPConfig()
-	FlpMetricsConfigDir = "test_metrics_definitions"
-	FlpMetricsConfig = TestFlpMetricsConfig
```
Comment (KalmanMeth) [inline controllers/flowlogspipeline/flp_test.go:594]: The idea here was to use a well-known (unchanging) set of metrics for testing, and not the current version in metrics_definitions, which might change from time-to-time. If we use the regular metrics_definitions, they whenever we make changes to metrics_definitions we may also have to update the test.

**Review by KalmanMeth — APPROVED** [review]

File: controllers/flowlogspipeline/flp_test.go (hunk 1 above)
Comment (jotak) [reply controllers/flowlogspipeline/flp_test.go:594]: Ok, but on the other hand, there's also some value in testing the actual embedded metrics (making sure it's correctly formed, etc.). Could be a bit more maintenance , but it also increases coverage

File: controllers/flowlogspipeline/flp_objects.go [hunk 2]
```diff
@@ -437,15 +413,21 @@ func (b *builder) buildPipelineConfig() ([]config.Stage, []config.StageParam) {
...
-func (b *builder) configMap() (*corev1.ConfigMap, string) {
-	stages, parameters := b.buildPipelineConfig()
+func (b *builder) configMap() (*corev1.ConfigMap, string, error) {
+	stages, parameters, err := b.buildPipelineConfig()
+	if err != nil {
+		return nil, "", err
+	}
 
 	config := map[string]interface{}{
 		"log-level": b.desired.LogLevel,
```
Comment (ronensc) [inline controllers/flowlogspipeline/flp_objects.go:433]: `config.ConfigFileStruct` can be used instead of `map[string]interface{}`

**Review by ronensc — APPROVED** [review]

**Review by mariomac — APPROVED** [review]

File: controllers/flowlogspipeline/flp_objects.go (hunk 2 above)
Comment (jotak) [reply controllers/flowlogspipeline/flp_objects.go:433]: hmm there are some weird things happening here: `ConfigFileStruct` doesn't have `health`. Seems like the output of confgen isn't exactly the input of flp ?

Comment (jotak) [reply controllers/flowlogspipeline/flp_objects.go:433]: also LogLevel exists in `ConfigFileStruct` but it seems unused (see `config.ParseConfig` => it doesn't fill log-level).
I guess some more cleanup should be done on FLP

Comment (ronensc) [reply controllers/flowlogspipeline/flp_objects.go:433]: Looking at the code of FLP's main, it seems to me that FLP accepts the health port only from the command line arguments.
https://github.com/netobserv/flowlogs-pipeline/blob/d7656a24bb2417bfd2fec25f75fb1ee077c1b86e/cmd/flowlogs-pipeline/main.go
If I'm correct, I think it's a bug in FLP that it doesn't allow the health port to be set from the config file.

Comment (jotak) [reply controllers/flowlogspipeline/flp_objects.go:433]: I think it works, as it uses viper which accepts a config file as a substitute to command line args

**Comment by jotak** [comment]
/approve
@ronensc I'm doing a second PR to try fixing some flaky tests, I'll use the filepath.Join as part of it

**Pull request merged**